// vuepress-plugin-meting/index.js
/*
 * @Author: 时不待我 790002517@qq.com
 * @Date: 2022-09-18 18:09:38
 * @LastEditors: 时不待我 790002517@qq.com
 * @LastEditTime: 2023-01-18 20:32:54
 */
const {
    resolve
  } = require('path')
  
  module.exports = (options = {}, context) => ({
    define() {
  
      const {
        metingApi = 'https://api.i-meto.com/meting/api',
          meting = {},
          aplayer = {},
          mobile = {},
      } = options
  
      const {
        auto = '', server = '', type = '', mid = ''
      } = meting
  
      const {
        additionalAudios = [],
          mini = null,
          autoplay = false,
          theme = '#b7daff',
          loop = 'all',
          order = 'list',
          preload = 'auto',
          volume = 0.7,
          customAudioType,
          mutex = true,
          lrcType = 0,
          listFolded = false,
          listMaxHeight = 250,
          storageName = 'vuepress-plugin-meting',
      } = aplayer
  
      const {
        cover = true, lrc = true
      } = mobile
  /* 加入 */
      let base = context.base || '/';
      aplayer.additionalAudios.forEach((element, index) => {
        let url = element.url;
        if (url.substr(0, 1) == '/') {
          url = url.substr(1);
          aplayer.additionalAudios[index].url = base + url
        }
      });
  /* 结束 */
      return {
        METING_API: metingApi,
        METING_OPTIONS: {
          auto,
          server,
          type,
          mid,
        },
        APLAYER_OPTIONS: {
          additionalAudios,
          mini,
          autoplay,
          theme,
          loop,
          order,
          preload,
          volume,
          customAudioType,
          mutex,
          lrcType,
          listFolded,
          listMaxHeight,
          storageName,
        },
        MOBILE_OPTIONS: {
          cover,
          lrc,
        },
      }
    },
  
    enhanceAppFiles: resolve(__dirname, './lib/enhanceAppFile.js'),
    globalUIComponents: options.meting !== undefined ? ['MetingGlobal'] : undefined,
  })
